library/ecdsa.c

```diff
@@ -278,10 +278,71 @@ int mbedtls_ecdsa_verify( mbedtls_ecp_group *grp,
     return( ret );
 }
 
+/*
+ * Convert a signature to a raw concatenation of {r, s}
+ */
+int mbedtls_ecdsa_signature_to_raw( const unsigned char *sig,
```

Comment:
This function copies some code from `mbedtls_ecdsa_read_signature`. Please factor the common code into a `static` function.